### DISCUSSION TITLE
x-mask $money wrongly adds comma and removes decimal point when pressing a letter key as input

### DISCUSSION THREAD

**Discussion by lilotop** [alpinejs/alpine#2934] [7. Bugs (archived)]
# Reproduction 1:

1. Create a money masked input as described [here](https://alpinejs.dev/plugins/mask#money-inputs).
2. Type **exactly 3** digits.
3. Try to type any non-digit character (e.g. `q`).
4. A comma (thousands separator) appears although the number is less than `1,000`.

![image](https://user-images.githubusercontent.com/5829157/168489114-b957d6bb-54e3-47a4-a0ac-bf76a34875de.png)

# Reproduction 2:

1. Create a money masked input as described [here](https://alpinejs.dev/plugins/mask#money-inputs).
2. Type a decimal value with **3 digits** before the decimal point (e.g. `123.42`).
3. Place the input cursor back in the beginning of the input (by using the mouse or arrow keys).
4. Try to type any non-digit character (e.g. `q`) several times.
5. On each character typed, the formatted number is morphed into some other buggy input, see below.

![image](https://user-images.githubusercontent.com/5829157/168489450-fb8f9a67-94f6-4e2f-abdc-ebd257dba7b1.png)

**Comment by SimoTod** [comment]
FYI, I believe they have been fixed [here](https://github.com/alpinejs/alpine/pull/2876/files) but the maintainer hasn't tagged a new release yet.
